nexus/db-queries/src/db/queries/vpc_subnet.rs

```diff
@@ -404,6 +410,81 @@ mod test {
             "Should be able to insert a VPC Subnet with the same ranges in a different VPC",
         );
 
+        // We shouldn't be able to insert a subnet if the IPv4 or IPv6 blocks overlap.
+        // Explicitly test for different CIDR masks with the same network address
+        let new_row = VpcSubnet::new(
+            other_other_subnet_id,
+            vpc_id,
+            make_id(&other_name, &description),
+            overlapping_ipv4_block_longer,
+            other_ipv6_block,
+        );
+        let err = db_datastore
+            .vpc_create_subnet_raw(new_row)
+            .await
+            .expect_err("Should not be able to insert VPC Subnet with overlapping IPv4 range {overlapping_ipv4_block_longer}");
+        assert_eq!(
+            err,
+            InsertVpcSubnetError::OverlappingIpRange(
+                overlapping_ipv4_block_longer.into()
+            ),
+            "InsertError variant should indicate an IP block overlaps"
+        );
```

Comment:
Smalll nit, since this is a test, but you could factor this into a function or closure, and then call it for both `overlapping_ipv4_block_{longer,shorter}`. Might just make this a bit cleaner.

Reply:
Talked to Ben offline about this.  These test cases are pretty small in size (2 statements) with a fair amount of variability into what they expect/assert, and it's not clear how to reformat this in a way that would actually reduce clutter.  Ben agreed we can go ahead without these changes.